### PR TITLE
Feat/dapp 536

### PR DESCRIPTION
**Pull request by mendesmcmg**
# Description

- Adds an onboarding modal that allows for choosing cause

# Necessary changes
- adds causes context

## Does this pull request close any open issues?

- [dapp-536](https://ribon.atlassian.net/browse/DAPP-536?atlOrigin=eyJpIjoiNjdmNGU3MGQ2ODYyNGQ1OWEzYWVkOGY3YzU0OWUwZGEiLCJwIjoiaiJ9)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Dangerous changes
- [ ] Chore (change that does not effect how the code works, eg: change color)
- [ ] Tests

## How to test

- Open anonymous tab an check for the third modal on the onboarding. The modal must allow you to choose a cause and filter nonProfits by the chosen cause.

### CODE REVIEW

File: src/contexts/causesContext/index.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  const [selectedCauseIndex, setSelectedCauseIndex] = useState(0);
+  const [causeIdSelectedByModal, setCauseIdSelectedByModal] = useState(
+    causeWasNotSelectedByModal,
+  );
+
+  const causesFilter = () => causes.filter((cause) => cause.active);
+
+  useEffect(() => {
+    setActiveCauses(causesFilter());
+    setCauseIdSelectedByModal(activeCauses[0]?.id);
```
Comment (NickNish09) [inline src/contexts/causesContext/index.tsx:36]: maybe we could call this `currentCauseId`, something more generic that could be used in other contexts besides the Modal.
What do you think?

File: src/pages/donations/CausesPage/ChooseCauseModal/CauseImage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      <S.ImageContainer src={coverImage} alt={name} />
+      <S.CauseName>{name}</S.CauseName>
+      <S.Intersection src={Intersection} />
+    </S.Container>
+  );
+}
+
+export default CauseImage;
+
+CauseImage.defaultProps = {
```
Comment (NickNish09) [inline src/pages/donations/CausesPage/ChooseCauseModal/CauseImage/index.tsx:31]: Do we need this defaultProps? If so we could think about a config that doesn`t make this required.
What do you think?

File: src/pages/donations/CausesPage/index.tsx [hunk 3]
```diff
@@ -111,6 +117,11 @@ function CausesPage(): JSX.Element {
     if (state?.failedDonation) logEvent("donateDonationError_view");
   }, []);
 
+  useEffect(() => {
```
Comment (NickNish09) [inline src/pages/donations/CausesPage/index.tsx:120]: Do we still need this `selectedCauseIndex` after the changes?

File: src/contexts/causesContext/index.tsx (hunk 1 above)
Comment (mendesmcmg) [reply src/contexts/causesContext/index.tsx:36]: right! good idea

File: src/pages/donations/CausesPage/ChooseCauseModal/CauseImage/index.tsx (hunk 2 above)
Comment (mendesmcmg) [reply src/pages/donations/CausesPage/ChooseCauseModal/CauseImage/index.tsx:31]: Yeeah it wasn't necessary here, but the linter was bugging about it :/

File: src/pages/donations/CausesPage/index.tsx (hunk 3 above)
Comment (mendesmcmg) [reply src/pages/donations/CausesPage/index.tsx:120]: you're rightt I removed it now

File: src/pages/donations/CausesPage/index.tsx [hunk 4]
```diff
@@ -167,19 +169,30 @@ function CausesPage(): JSX.Element {
...
-        const newCauseIndex = causesFilter().findIndex(
+        const newCauseIndex = activeCauses.findIndex(
           (cause) => cause.id === currentNonProfit.cause.id,
         );
         setSelectedButtonIndex(newCauseIndex);
       }
     }
   }, [currentNonProfitIndex]);
 
+  useEffect(() => {
```
Comment (leodhb) [inline src/pages/donations/CausesPage/index.tsx:183]: ONBOARDING: change nonprofit index when user selects a different cause on modal

**Review by leodhb — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**